jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -110,15 +115,19 @@ class S3TreeTable(
             S3Telemetry.downloadObject(project, false)
             return true
         }
-        val fileWrapper = VirtualFileWrapper(File("${FileUtil.getTempDirectory()}${File.separator}${objectNode.key.replace('/', '_')}"))
+        val versionPostfix = if (objectNode is S3TreeObjectVersionNode) "_${objectNode.versionId}" else ""
+        val fileWrapper = VirtualFileWrapper(File("${FileUtil.getTempDirectory()}${File.separator}${objectNode.key.replace('/', '_')}$versionPostfix"))
```

Comment:
is it possible that with the addition of the version postfix, we can exceed the windows max filename length?

Reply:
Good point but I think we are still safe, `versionId` adds 32 characters on top of filename so it's still long way to `255` limit.